l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -204,18 +210,27 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
     function registerSettlementLayer(
         uint256 _newSettlementLayerChainId,
         bool _isWhitelisted
-    ) external onlyChainSTM(_newSettlementLayerChainId) {
+    ) external onlyChainSTM(_newSettlementLayerChainId) onlyL1 {
         whitelistedSettlementLayers[_newSettlementLayerChainId] = _isWhitelisted;
-
-        // TODO: emit event
+        emit SettlementLayerRegistered(_newSettlementLayerChainId, _isWhitelisted);
     }
 
     /// @dev Used to set the assetAddress for a given assetInfo.
     /// @param _additionalData the additional data to identify the asset
     /// @param _assetAddress the asset handler address
-    function setAssetHandlerAddressInitial(bytes32 _additionalData, address _assetAddress) external {
-        address sender = L1_CHAIN_ID == block.chainid ? msg.sender : AddressAliasHelper.undoL1ToL2Alias(msg.sender); // Todo: this might be dangerous. We should decide based on the tx type.
-        bytes32 assetInfo = keccak256(abi.encode(L1_CHAIN_ID, sender, _additionalData)); /// todo make other asse
+    function setAssetHandlerAddress(bytes32 _additionalData, address _assetAddress) external {
+        // It is a simplified version of the logic used by the AssetRouter to manage asset handlers.
+        // STM's assetId is `keccak256(abi.encode(L1_CHAIN_ID, stmDeployer, stmAddress))`.
+        // And the STMDeployer is considered the deployment tracker for the STM asset.
+        //
+        // The STMDeployer will call this method to set the asset handler address for the assetId.
+        // If the chain is not the same as L1, we assume that it is done via L1->L2 communication and so we unalias the sender.
+        //
+        // For simpler handling we allow anyone to call this method. It is okay, since during bridging operations
```

Comment:
Should we mark in as a comment that this needs to be reviewed once interop happens?